src/components/SettingsBarContext.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, ReactNode, useState } from 'react';
+import { FeatureStatus } from '../models';
+
+import api from '../service/api';
+
+export type HistoryBarContextType = {
+  handleSetBar: () => void,
+  handleUpdateFavotites: () => void,
+  handleUpdateHistory: () => void,
+  historyBar: FeatureStatus,
+  favoriteContextState: number,
+  historyContextState: number,
+}
+
+export const HistoryBarContext = createContext<HistoryBarContextType>({
+  handleSetBar: () => { },
+  handleUpdateFavotites: () => { },
+  handleUpdateHistory: () => { },
+  historyBar: FeatureStatus.disabled,
+  favoriteContextState: 0,
+  historyContextState: 0,
+});
+
+type HistoryBarProviderType = {
+  children?: ReactNode
+}
+
+export const HistoryBarProvider = ({ children }: HistoryBarProviderType) => {
+  const [historyBar, setHistoryBar] = useState(api.getStore().historyBar);
+  const [favoriteContextState, setFavoriteContextState] = useState(api.getFavoritsIdList().length);
+  const [historyContextState, setHistoryContextState] = useState(api.getHistoryIdList().length);
+
+  const handleSetBar = () => {
+    setHistoryBar(api.switchHistoryBar());
+  }
+
+  const handleUpdateFavotites = () => {
+    setFavoriteContextState(api.getFavoritsIdList().length);
+  }
+
+  const handleUpdateHistory = () => {
+    setHistoryContextState(api.getHistoryIdList().length)
+  }
+
+  return (
+    <HistoryBarContext.Provider value={{
+      handleSetBar, handleUpdateFavotites, handleUpdateHistory,
```

Comment:
How HistoryBar connected to Favorites?

Reply:
I use HistoryBar to allow or not to add new last seen and HistoryBar saved in SettingsBarContext with other pieces of settings bar context